A/AWSLC/build_tarballs.jl

```diff
@@ -0,0 +1,48 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "AWSLC"
+version = v"1.0.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/awslabs/aws-lc.git", "11b50d39cf2378703a4ca6b6fee9d76a2e9852d1")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DBUILD_STATIC_LIBS=OFF -DBUILD_SHARED_LIBS=ON -GNinja aws-lc.git
+ninja -j${nproc}
+ninja install
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl")
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libcrypto", :libcrypto),
+    LibraryProduct("libssl", :libssl),
```

Comment:
These will conflict with OpenSSL.....

Reply:
I will rename them

Reply:
Is there no way to support AWS-LC, then? What prevents me from renaming these?

Reply:
> Is there no way to support AWS-LC, then?

I didn't say that :slightly_smiling_face:  It's just that if you have OpenSSL and AWS-LC in the same environment, chaos ensues.

> What prevents me from renaming these?

Renaming libraries isn't that easy.  If you really wanted to do that, you'd also have to change their sonames.  But then downstream libraries won't like it.  See for example #4725 for the kind of mess renaming libraries causes.

Reply:
I only need AWS-LC because it's a dependency of [aws-c-mqtt](https://github.com/awslabs/aws-c-mqtt). Maybe I could build AWS-LC directly with aws-c-mqtt instead of as a separate library?

Reply:
Uhm, doing static linking would actually help here.